lib/auth/v4/headerAuthCheck.js

```diff
@@ -34,18 +36,15 @@ headerAuthCheck.check = (request, log, callback, awsService) => {
         return callback(errors.MissingSecurityHeader);
     }
 
-    let payloadChecksum;
-    if (!awsService || awsService === 's3' || awsService === 'iam') {
-        payloadChecksum = request.headers['x-amz-content-sha256'];
-        if (!payloadChecksum) {
-            log.warn('missing payload checksum');
-            return callback(errors.MissingSecurityHeader);
-        }
-        if (payloadChecksum === 'STREAMING-AWS4-HMAC-SHA256-PAYLOAD') {
-            log.trace('requesting streaming v4 auth');
-            // TODO: Implement this option
-            return callback(errors.NotImplemented);
-        }
+    const payloadChecksum = request.headers['x-amz-content-sha256'];
+    if (!payloadChecksum) {
+        log.warn('missing payload checksum');
```

Comment:
Well, since that's a parameter entirely reliant on the client, it's a potential attack vector.